school_center.py

```diff
@@ -52,6 +55,17 @@ def sort_key(c):
     within_distance = []
     nearest_distance = None;
     nearest_center = None
+
+     # Use distance cache
+    cache_key = (school_lat, school_long)
+    if cache_key in distance_cache:
+        distances = distance_cache[cache_key]
+    else:
+        distances = {c['cscode']: haversine_distance(
```

Comment:
the way schools are iterated later, dont think there will ever be a cache hit when keys are organized this way.